aiida_quantumespresso/parsers/raw_parser_pw.py

```diff
@@ -1698,6 +1704,42 @@ def parse_pw_text_output(data, xml_data={}, structure_data={}, input_dict={}, pa
                 except Exception:
                     parsed_data['warnings'].append('Error while parsing stress tensor.')
 
+            # Electronic and ionic dipoles when 'lelfield' was set to True in input parameters
+            elif lelfield is True:
+
+                if 'Electronic Dipole per cell' in line:
+                    electronic_dipole = float(line.split()[-1])
+                    trajectory_frame.setdefault('electronic_dipole_cell_average', []).append(electronic_dipole)
+
+                elif 'Ionic Dipole per cell' in line:
+                    ionic_dipole = float(line.split()[-1])
+                    trajectory_frame.setdefault('ionic_dipole_cell_average', []).append(ionic_dipole)
+
+                elif 'Electronic Dipole on Cartesian axes' in line:
+                    electronic_dipole = [float(data_step[count + i + 1].split()[1]) for i in range(3)]
+                    trajectory_frame.setdefault('electronic_dipole_cartesian_axes', []).append(electronic_dipole)
+
+                elif 'Ionic Dipole on Cartesian axes' in line:
+                    ionic_dipole = [float(data_step[count + i + 1].split()[1]) for i in range(3)]
+                    trajectory_frame.setdefault('ionic_dipole_cartesian_axes', []).append(ionic_dipole)
+
+
+        # End of trajectory frame, only keep last entries for dipole related values
+        if lelfield is True:
+
+            # For every property only get the last entry if possible
+            ed_cell = trajectory_frame['electronic_dipole_cell_average'].pop()
```

Comment:
If by any reason none is found (e.g. 0 steps, or the syntax changes and the lines are not found anymore) this would crash I think... can we pop with a default value?